hles/extraction/src/main/scala/org/broadinstitute/monster/dap/ExtractionPipelineBuilder.scala

```diff
@@ -81,25 +85,28 @@ class ExtractionPipelineBuilder(
     // NOTE: This logic is replicated from the encode-ingest pipeline,
     // we should consider moving it to scio-utils.
     val batchedIds = idsToExtract
+      // Pack up the IDs into dummy objects so beam knows how to handle them,
+      // then encode them into UTF8 so they make consistent JSON.
       .map(KV.of("", _))
       .setCoder(KvCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of()))
+      // Batch up the IDs
       .applyKvTransform(GroupIntoBatches.ofSize(idBatchSize.toLong))
       .map { ids =>
+        // Construct a query to grab the full records for each batch of study IDs
         GetRecords(
           ids = ids.getValue.asScala.toList,
           forms = formsForExtraction,
-          // List of fields to pull out of the data for us to filter on
           fields = fieldList
         )
       }
 
-    // Download the form data for each batch of records.
+    // Flatten the batches of results into a single, massive list once they've come back
     val extractedRecords = batchedIds.transform("Get records") {
       _.applyKvTransform(ParDo.of(lookupFn)).flatMap(kv => kv.getValue.fold(throw _, _.arr))
```

Comment:
given how frequently this "grab batched results and flatten them to a list" line comes up, and how arcane it is, i think it'd benefit from being factored out into a common function, but i also don't think that's in scope for this PR.